### PR TITLE
Dev delete user

### PR DESCRIPTION
**Pull request by ian-dev**

### CODE REVIEW

File: src/components/Table.js [hunk 1]
```diff
@@ -1,22 +1,37 @@
 import { useMemo, useState } from "react";
 import { useGetUsersQuery } from "../utils/api";
 import { DataGrid } from "@mui/x-data-grid";
-import GenericButton from "./GenericButton";
-import VisibilityIcon from "@mui/icons-material/Visibility";
-import RemoveCircleIcon from "@mui/icons-material/RemoveCircle";
+import Button from "@mui/material/Button";
+import { useRef } from "react";
+import { useEffect } from "react";
```
Comment (nassao) [inline src/components/Table.js:6]: You can combine those two lines in a single import:
`import React, { useRef, useEffect } from "react";`

File: src/components/Table.js [hunk 2]
```diff
@@ -1,22 +1,37 @@
...
+  const userListHelper = useRef(USERS);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    userListHelper.current = data;
+  }, [data]);
+
+  // CLICK HANDLERS
+  const viewClickHandler = (e) => {
+    // debugger;
```
Comment (nassao) [inline src/components/Table.js:26]: Try to avoid commented code

File: src/pages/UserProfile.js [hunk 3]
```diff
@@ -1,53 +1,83 @@
+import { useParams } from "react-router-dom";
+import { useGetCurrentUserQuery } from "../utils/api";
+import { Card, CardHeader, CardContent } from "@mui/material";
 import GenericContainer from "../components/GenericContainer";
-import GenericButton from "../components/GenericButton";
 import UserAvatar from "../components/UserAvatar";
-import { Card, CardHeader, CardContent } from "@mui/material";
-import TextField from "@mui/material/TextField";
+// import TextField from "@mui/material/TextField";
+// import { useSelector } from "react-redux";
```
Comment (nassao) [inline src/pages/UserProfile.js:7]: Avoid commented code

File: src/pages/UserProfile.js [hunk 4]
```diff
@@ -1,53 +1,83 @@
...
+  //
+  /* const users = useSelector(
+    (state) => state.usersAdmin.queries["getUsers(undefined)"].data
+  );
+  debugger;
+  const user = users.find((user) => parseInt(user.id) === parseInt(userid)); */
+  //
+  //
+
+  console.log(data);
```
Comment (nassao) [inline src/pages/UserProfile.js:34]: Try not to include the loggers in the committed code

File: src/utils/server.js [hunk 5]
```diff
@@ -1,24 +1,23 @@
-import { rest } from 'msw';
-import mockUsers from './data';
+import { rest } from "msw";
```
Comment (nassao) [inline src/utils/server.js:1]: Use single quotes standard

**Review by nassao — CHANGES_REQUESTED** [review]
Always use the single quote standard

File: src/pages/UserProfile.js [hunk 6]
```diff
@@ -1,53 +1,83 @@
...
-        />
+        <CardHeader avatar={<UserAvatar image={"A"} />} title={data.name} />
         <CardContent>
+          <h3>Id: {data.id}</h3>
+          <h3>Name: {data.name}</h3>
+          <h3>Email: {data.email}</h3>
+          <h3>Gender: {data.gender}</h3>
+          <h3>Address: {data.address}</h3>
+          <h3>Status: {data.status}</h3>
+          <h3>UUID: {data.uuid}</h3>
```
Comment (santi1991) [inline src/pages/UserProfile.js:47]: use Material UI components, `import Typography from '@mui/material/Typography';`

File: src/utils/api.js [hunk 7]
```diff
@@ -20,6 +20,15 @@ export const usersApi = createApi({
     },
   }),
   endpoints: (build) => ({
+    getCurrentUser: build.query({
+      query: (params) => {
+        const { id } = params;
+        return {
```
Comment (santi1991) [inline src/utils/api.js:26]: do not forget to include the `params` property. query endpoints use it.

**Review by santi1991 — CHANGES_REQUESTED** [review]

File: src/utils/api.js (hunk 7 above)
Comment (santi1991) [reply src/utils/api.js:26]: @nassao

**Pull request merged**